src/fluree/db/query/exec/eval.cljc

```diff
@@ -117,16 +118,166 @@
   (str/ends-with? s substr))
 
 (defn subStr
-  [s start end]
-  (subs s start end))
+  ([s start]
+   (subStr s start (count s)))
+  ([s start length]
+   ;; The index of the first character in a string is 1.
+   (let [start (dec start)]
+     (subs s start (min (+ start length)
+                        (count s))))))
+
+(defn strLen
+  [s]
+  (count s))
+
+(defn ucase
+  [s]
+  (str/upper-case s))
+
+(defn lcase
+  [s]
+  (str/lower-case s))
+
+(defn contains
+  [s substr]
+  (str/includes? s substr))
+
+(defn strBefore
+  [s substr]
+  (-> (str/split s (re-pattern substr))
+      first
+      str))
+
+(defn strAfter
+  [s substr]
+  (-> (str/split s (re-pattern substr))
```

Comment:
It doesn't look like this matches the spec either. First, for the same reason as `strBefore` above, and also because [the spec says to return the string after the _first_ occurrence of `substr`](https://www.w3.org/TR/sparql11-query/#func-strafter), and this will return the string after the last occurrence of `substr`